engine/api/src/org/apache/cloudstack/engine/subsystem/api/storage/HypervisorHostListener.java

```diff
@@ -21,7 +21,13 @@
 import com.cloud.exception.StorageConflictException;
 
 public interface HypervisorHostListener {
+    boolean hostAdded(long hostId);
+
     boolean hostConnect(long hostId, long poolId) throws StorageConflictException;
 
     boolean hostDisconnected(long hostId, long poolId);
+
+    boolean hostAboutToBeRemoved(long hostId);
```

Comment:
This event was added specifically to address hypervisors like VMware that need to be told to remove iSCSI targets.

Those kind of hypervisor hosts need to be notified to do this before the host is removed because CloudStack can no longer talk to those hosts once they're in the Removed state.